feat(client): add onContentUpdated composition API

pengzhanbo opened the pull request:
<!-- Thank you for contributing! -->

### Before submitting the PR, please make sure you do the following <!-- (put an "X" next to an item) -->

- [x] Read the [Contributing Guidelines](https://github.com/vuepress/core/blob/main/CONTRIBUTING.md).
- [x] Provide a description in this PR that addresses **what** the PR is solving. If this PR is going to solve an existing issue, please reference the issue (e.g. `close #123`).

### What is the purpose of this pull request? <!-- (put an "X" next to an item) -->

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Other

### Description

When the markdown content changes, using the built-in Vue lifecycle hooks does not work well, so the `onContentUpdated` hook has been added to properly listen for changes in the content.

This applies to scenarios such as `medium-zoom` or `photo-swipe` that require reinitialization when the page content changes.

Mister-Hope commented:
uepress/docs also needs a PR

pengzhanbo commented:
> uepress/docs also needs a PR

<https://github.com/vuepress/docs/pull/44>

meteorlxy reviewed (changes requested):
e2e tests are required

pengzhanbo commented:
@meteorlxy @Mister-Hope 
I have rewritten this PR and added relevant e2e tests referencing hmr, but it is not working properly in vite. Can you help me resolve this?

meteorlxy commented on packages/client/src/composables/onContentUpdated.ts:
```diff
@@ -0,0 +1,21 @@
+  })
+}
+
+export function runCallbacks(reason: ContentUpdatedReason): void {
```
- Should use arrow function to keep consistent with other functions
- The name could be improved.
- Add js docs. Maybe a `@internal` mark is better.

meteorlxy commented on packages/client/src/composables/index.ts:
```diff
@@ -1,3 +1,4 @@
 export * from './clientData.js'
 export * from './clientDataUtils.js'
 export * from './updateHead.js'
+export * from './onContentUpdated.js'
```
Reorder the exports in alphabetically
